test/integration/framework/main.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2019 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"k8s.io/client-go/discovery"
+	"k8s.io/klog"
+
+	dynamicdiscovery "metacontroller.app/dynamic/discovery"
+	"metacontroller.app/server"
+)
+
+var resourceMap *dynamicdiscovery.ResourceMap
+
+// TestMain starts etcd, kube-apiserver, and metacontroller before running tests.
+func TestMain(tests func() int) {
+	var result int
+	defer func() {
+		os.Exit(result)
+	}()
+
+	stopEtcd, err := startEtcd()
+	if err != nil {
+		klog.Fatalf("cannot run integration tests: unable to start etcd: %v", err)
+	}
+	defer stopEtcd()
+
+	stopApiserver, err := startApiserver()
+	if err != nil {
+		klog.Fatalf("cannot run integration tests: unable to start kube-apiserver: %v", err)
+	}
+	defer stopApiserver()
+
+	klog.Info("Waiting for kube-apiserver to be ready...")
+	start := time.Now()
+	for {
+		if err := execKubectl("version"); err == nil {
+			break
+		}
+		if time.Since(start) > defaultWaitTimeout {
+			klog.Fatalf("timed out waiting for kube-apiserver to be ready: %v", err)
+		}
+		time.Sleep(time.Second)
+	}
+
+	// Install Metacontroller RBAC.
+	if err := execKubectl("apply", "-f", "../../../manifests/metacontroller-rbac.yaml"); err != nil {
```

Comment:
Should we set `../../../manifests/` as a global variable? It looks like we may want to add more manifests to the path, and the path will be called more frequently.